### DISCUSSION TITLE
Allow PCSX-Redux users to add memory watchpoints

### DISCUSSION THREAD

**Discussion by ghost** [grumpycoders/pcsx-redux#1071] [Ideas]
A memory watchpoint is a thing that allows the user to see the value of some byte, half or word in the address that the watchpoint was added.

Add a new Watch window in the Debug menu where the user can add watchpoints and view the values of some bytes, halfs and words in the RAM of the emulated rom.

The Watch window contains a ListView with three columns: "Description", "Address" and "Size" that is always either "Byte", "Half" or "Word".

Additionally add to the context menu of the Memory Editor a new command "Add Watchpoint"

When the user chooses "Add Watchpoint" from the context menu of the Memory Editor a new dialog appears asking the user to describe the watchpoint (give it a description), choose the size of the watchpoint which can be either Byte, Half or Word and whether or not the value of the watching byte, half or word is signed or unsigned, display as hex or decimal or 32 bit floating number.

**Comment by nicolasnoble** [comment]
Same as with the other discussion, you can write to memory from Lua, create data breakpoints, and you can even create UI.

Check https://pcsx-redux.consoledev.net/lua/#spyro-year-of-the-dragon

**Reply by ghost** [reply]
> Same as with the other discussion, you can write to memory from Lua, create data breakpoints, and you can even create UI.
> 
> Check https://pcsx-redux.consoledev.net/lua/#spyro-year-of-the-dragon

I will learn this Lua script and this feature request can also receive a low priority too.

**Comment by wheremyfoodat** [comment]
In addition to using the already implemented data breakpoints and the LUA API, you can also "watch" a value in RAM in realtime by using the memory observer widget. Ie you can see how and when it changes.

**Reply by ghost** [reply]
> In addition to using the already implemented data breakpoints and the LUA API, you can also "watch" a value in RAM in realtime by using the memory observer widget. Ie you can see how and when it changes.

Can I add an address manually in this memory observer window? Because I don't see any "Add an address manually" button there in any of his tabs.
